maven-plugin/src/main/java/io/confluent/kafka/schemaregistry/maven/derive/schema/DeriveProtobufSchema.java

```diff
@@ -0,0 +1,157 @@
+    classToDataType.put(com.fasterxml.jackson.databind.node.MissingNode.class.getName(), ANY_FIELD);
+  }
+
+  /**
+   * Merge different records into one record and merge number types: double and int/long
+   * If there are multiple data types or nested arrays error is returned
+   */
+  protected ObjectNode mergeMultipleDataTypes(ObjectNode mergedArray,
+                                              List<JsonNode> primitives,
+                                              List<JsonNode> records,
+                                              List<JsonNode> arrays,
+                                              boolean check2dArray) {
+    ArrayNode items = mapper.createArrayNode();
+    if (arrays.size() > 0 && check2dArray) {
+      throw new IllegalArgumentException(String.format("Found nested array: %s", arrays));
+    } else if (primitives.stream().anyMatch(o -> o.get("type").asText().equals(ANY_FIELD))) {
+      throw new IllegalArgumentException("Repeated field elements cannot be null");
+    }
+
+    DeriveSchemaUtils.mergeNumberTypes(primitives);
+    items.addAll(DeriveSchemaUtils.getUnique(primitives));
+    // No merging of arrays, add directly to items' list
+    items.addAll(arrays);
+
+    if (records.size() > 0) {
+      items.add(mergeRecords(records));
+    }
+
+    if (items.size() > 1) {
+      throw new IllegalArgumentException(String.format("Found multiple data types: %s", items));
+    } else if (items.size() == 1) {
+      mergedArray.set("items", items.get(0));
+    } else {
+      mergedArray.set("items", getNullSchema());
+    }
+    return mergedArray;
+  }
+
+  protected void checkName(String name) {
+    if (name.isEmpty()) {
+      throw new IllegalArgumentException("Name cannot be empty");
+    }
+    if (!name.matches("[a-zA-Z\\d-._]+")) {
+      throw new IllegalArgumentException("Name must only contain alphanumerics "
+          + "or one of \"-\", \"_\" and \".\" ");
+    }
+    if (Character.isDigit(name.charAt(0))) {
+      throw new IllegalArgumentException("Name cannot begin with a digit");
+    }
+  }
+
+  /**
+   * Converts json schema template to protobuf format
+   */
+  @Override
+  protected TextNode convertToFormat(JsonNode schema, String name) {
+    String schemaForRecord = convertToFormatRecord(schema, name);
+    StringBuilder schemaBuilder = new StringBuilder(PROTOBUF_SYNTAX);
+    if (schemaForRecord.contains(ANY_FIELD)) {
+      schemaBuilder.append(IMPORT_ANY_FIELD);
+    }
+    schemaBuilder.append(schemaForRecord);
+    ProtobufSchema protobufSchema = new ProtobufSchema(schemaBuilder.toString());
+    protobufSchema.validate();
+    return mapper.convertValue(protobufSchema.toString(), TextNode.class);
+  }
+
+  protected String convertToFormatRecord(JsonNode schema, String name) {
+    int fieldNum = 1;
+    StringBuilder protobufSchema = new StringBuilder();
+    protobufSchema.append(String.format("message %s { %n", name));
```

Comment:
we have two parameters, but one value is assigned. Is the last one supposed to be \n